activerecord/lib/active_record/persistence.rb

```diff
@@ -971,11 +972,16 @@ def _update_record(attribute_names = self.attribute_names)
     def _create_record(attribute_names = self.attribute_names)
       attribute_names = attributes_for_create(attribute_names)
 
-      new_id = self.class._insert_record(
+      result = self.class._insert_record(
         attributes_with_values(attribute_names)
       )
 
-      self.id ||= new_id if @primary_key
+      if result.is_a?(Hash)
+        self.id ||= result["id"] if @primary_key
+        self.assign_attributes(result.except("id"))
```

Comment:
Why do in two lines what looks like only needs one?

Reply:
I can check if keeping only the `assign_attributes` call can work, wanted to stay close to current implementation